barretenberg/cpp/src/barretenberg/ultra_honk/oink_verifier.cpp

```diff
@@ -0,0 +1,139 @@
+#include "barretenberg/ultra_honk/oink_verifier.hpp"
+
+namespace bb {
+
+/**
+ * @brief Oink Verifier function that runs all the rounds of the verifier
+ * @details Returns the witness commitments and relation_parameters
+ * @tparam Flavor
+ * @return OinkOutput<Flavor>
+ */
+template <IsUltraFlavor Flavor> OinkOutput<Flavor> OinkVerifier<Flavor>::verify()
+{
+    // Execute the Verifier rounds
+    execute_preamble_round();
+    execute_wire_commitments_round();
+    execute_sorted_list_accumulator_round();
+    execute_log_derivative_inverse_round();
+    execute_grand_product_computation_round();
+
+    return OinkOutput<Flavor>{
+        .relation_parameters = relation_parameters,
+        .commitments = witness_comms,
+    };
+}
+
+/**
+ * @brief Get circuit size, public input size, and public inputs from transcript
+ *
+ */
+template <IsUltraFlavor Flavor> void OinkVerifier<Flavor>::execute_preamble_round()
+{
+    // TODO(Adrian): Change the initialization of the transcript to take the VK hash?
+    const auto circuit_size = transcript->template receive_from_prover<uint32_t>(domain_separator + "circuit_size");
+    const auto public_input_size =
+        transcript->template receive_from_prover<uint32_t>(domain_separator + "public_input_size");
+    const auto pub_inputs_offset =
+        transcript->template receive_from_prover<uint32_t>(domain_separator + "pub_inputs_offset");
+
+    ASSERT(circuit_size == key->circuit_size);
+    ASSERT(public_input_size == key->num_public_inputs);
+    ASSERT(pub_inputs_offset == key->pub_inputs_offset);
```

Comment:
probably not needed as well. This should just be overwritten by transcript pub_inputs_offset?